proto-lens/src/Data/ProtoLens/Encoding/Growing.hs

```diff
@@ -0,0 +1,59 @@
+-- | A mutable vector that grows in size.
+module Data.ProtoLens.Encoding.Growing (
+    Growing,
+    new,
+    append,
+    unsafeFreeze,
+    RealWorld,
+    ) where
+
+import Control.Monad.Primitive (PrimMonad, PrimState, RealWorld)
+
+import qualified Data.Vector.Generic as V
+import qualified Data.Vector.Generic.Mutable as MV
+
+-- | A mutable vector which can increase in capacity.
+data Growing v s a = Growing
+    {-# UNPACK #-} !Int
+        -- The number of elements in the mutable vector
```

Comment:
Turn into a haddock field comment?

Reply:
It breaks on older versions of Haddock due to the bang pattern.

Since the constructor isn't exposed, it won't show in the generated html so I don't think it matters in practice.